cmake/BLTPrivateMacros.cmake

```diff
@@ -754,6 +754,24 @@ macro(blt_add_hip_executable)
 
 endmacro(blt_add_hip_executable)
 
+##-----------------------------------------------------------------------------
+## blt_error_if_target_exists(<extension_list> <regex_output_variable>)
+##
+## This function converts the list of extensions in <extension_list>
+## to a regular expression in <regex_output_variable> that should match
+## any file name starting with a string and ending with any one of the
+## extensions in <extension_list>.
+##
+## Note that the <extension_list> argument has to be quoted, so that the
+## argument is passed as a string to this function.
+## -----------------------------------------------------------------------------
+function(blt_make_regex_from_ext_list ext_list ext_regex)
+  message(STATUS "ext_list = ${ext_list}")
```

Comment:
This would like be too verbose for typical usage. 

Once we're satisfied with the functionality, please consider removing/commenting this out, or adding a verbose flag to the function/macro to have it be quiet by default.